src/plugins/presentation_panel/public/panel_component/presentation_panel_internal.tsx

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { EuiFlexGroup, EuiPanel, htmlIdGenerator } from '@elastic/eui';
+import classNames from 'classnames';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Subscription } from 'rxjs';
+
+import { PanelLoader } from '@kbn/panel-loader';
+import { apiFiresPhaseEvents, useBatchedPublishingSubjects } from '@kbn/presentation-publishing';
+import { PresentationPanelHeader } from './panel_header/presentation_panel_header';
+import { PresentationPanelError } from './presentation_panel_error';
+import { DefaultPresentationPanelApi, PresentationPanelInternalProps } from './types';
+
+export const PresentationPanelInternal = <
+  ApiType extends DefaultPresentationPanelApi = DefaultPresentationPanelApi,
+  ComponentPropsType extends {} = {}
+>({
+  index,
+  hideHeader,
+  showShadow,
+
+  showBadges,
+  showNotifications,
+  getActions,
+  actionPredicate,
+
+  Component,
+  componentProps,
+
+  onPanelStatusChange,
+}: PresentationPanelInternalProps<ApiType, ComponentPropsType>) => {
+  const [api, setApi] = useState<ApiType | null>(null);
+  const headerId = useMemo(() => htmlIdGenerator()(), []);
+
+  const {
+    uuid,
+    viewMode,
+    blockingError,
+    panelTitle,
+    dataLoading,
+    hidePanelTitle,
+    panelDescription,
+    defaultPanelTitle,
+    parentHidePanelTitle,
+  } = useBatchedPublishingSubjects({
+    dataLoading: api?.dataLoading,
+    blockingError: api?.blockingError,
+    viewMode: api?.viewMode,
+    uuid: api?.uuid,
+
+    panelTitle: api?.panelTitle,
+    hidePanelTitle: api?.hidePanelTitle,
+    panelDescription: api?.panelDescription,
+    defaultPanelTitle: api?.defaultPanelTitle,
+    parentHidePanelTitle: (api?.parentApi?.value as DefaultPresentationPanelApi)?.hidePanelTitle,
+  });
+
+  const [initialLoadComplete, setInitialLoadComplete] = useState(!dataLoading);
+  if (dataLoading === false && !initialLoadComplete) {
+    setInitialLoadComplete(true);
```

Comment:
Should setInitialLoadComplete be inside an effect. State should not be set from rendering.

Reply:
This code was just copied over from https://github.com/elastic/kibana/pull/171238. Refer to [this comment](https://github.com/elastic/kibana/pull/171238/files#r1403113178)  for more info on why it was done this way. I'd be okay with wrapping this in a `useEffect`.